MinPlatformPkg/Test/Library/TestPointLib/StandaloneMmTestPointCommunication.c

```diff
@@ -0,0 +1,379 @@
+  @param CommBufferSize  The size of the CommBuffer.
+
+  @retval EFI_SUCCESS Command is handled successfully.
+**/
+EFI_STATUS
+EFIAPI
+StandaloneMmTestPointSmiHandler (
+  IN EFI_HANDLE  DispatchHandle,
+  IN CONST VOID  *Context         OPTIONAL,
+  IN OUT VOID    *CommBuffer      OPTIONAL,
+  IN OUT UINTN   *CommBufferSize  OPTIONAL
+  )
+{
+  SMI_HANDLER_TEST_POINT_PARAMETER_HEADER     *SmiHandlerTestPointParameterHeader;
+  UINTN                                       TempCommBufferSize;
+
+  DEBUG((DEBUG_ERROR, "StandaloneMmTestPointSmiHandler Enter\n"));
+
+  // PROBLEM HERE FOR TESTPOINT
+
+  //
+  // If input is invalid, stop processing this SMI
+  //
+  if (CommBuffer == NULL || CommBufferSize == NULL) {
+    return EFI_SUCCESS;
+  }
+
+  TempCommBufferSize = *CommBufferSize;
+
+  if (TempCommBufferSize < sizeof(SMI_HANDLER_TEST_POINT_PARAMETER_HEADER)) {
+    DEBUG((DEBUG_ERROR, "StandaloneMmTestPointSmiHandler: StandaloneMm communication buffer size invalid!\n"));
+    return EFI_SUCCESS;
+  }
+
+  if (!MmCommBufferValid((UINTN)CommBuffer, TempCommBufferSize)) {
+    DEBUG((DEBUG_ERROR, "StandaloneMmTestPointSmiHandler: StandaloneMm communication buffer in SMRAM or overflow!\n"));
+    return EFI_SUCCESS;
+  }
+
+  SmiHandlerTestPointParameterHeader = (SMI_HANDLER_TEST_POINT_PARAMETER_HEADER *)((UINTN)CommBuffer);
+  SmiHandlerTestPointParameterHeader->ReturnStatus = (UINT64)-1;
+
+  switch (SmiHandlerTestPointParameterHeader->Command) {
+  case SMI_HANDLER_TEST_POINT_COMMAND_GET_INFO:
+    DEBUG((DEBUG_ERROR, "SmiHandlerTestPointHandlerGetInfo\n"));
+    if (TempCommBufferSize != sizeof(SMI_HANDLER_TEST_POINT_PARAMETER_GET_INFO)) {
+      DEBUG((DEBUG_ERROR, "StandaloneMmTestPointSmiHandler: StandaloneMm communication buffer size invalid!\n"));
+      return EFI_SUCCESS;
+    }
+    StandaloneMmTestPointSmiHandlerGetInfo((SMI_HANDLER_TEST_POINT_PARAMETER_GET_INFO *)(UINTN)CommBuffer);
+    break;
+  case SMI_HANDLER_TEST_POINT_COMMAND_GET_DATA_BY_OFFSET:
+    DEBUG((DEBUG_ERROR, "SmiHandlerTestPointHandlerGetDataByOffset\n"));
+    if (TempCommBufferSize != sizeof(SMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET)) {
+      DEBUG((DEBUG_ERROR, "StandaloneMmTestPointSmiHandler: StandaloneMm communication buffer size invalid!\n"));
+      return EFI_SUCCESS;
+    }
+    StandaloneMmTestPointSmiHandlerGetDataByOffset((SMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET *)(UINTN)CommBuffer);
+    break;
+  default:
+    break;
+  }
+
+  DEBUG((DEBUG_ERROR, "StandaloneMmTestPointSmiHandler Exit\n"));
+
+  return EFI_SUCCESS;
+}
+
+/**
+  Register StandaloneMm TestPoint handler.
+**/
+VOID
+RegisterStandaloneMmTestPointSmiHandler (
+  VOID
+  )
+{
+  EFI_HANDLE                       DispatchHandle;
+  EFI_STATUS                       Status;
+  STATIC BOOLEAN                   Registered = FALSE;
+
+  if (Registered) {
+    return ;
+  }
+  
+  Status = gMmst->MmiHandlerRegister (
+                    StandaloneMmTestPointSmiHandler,
```

Comment:
Not a big deal but it should technically be called an MMI handler instead of SMI handler here, so `StandaloneMmTestPointMmiHandler`.

Reply:
Fixed.